build/yamls/patches/eks/eksEnv.yml

```diff
@@ -0,0 +1,14 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: antrea-agent
+spec:
+  template:
+    spec:
+      containers:
+        - name: antrea-agent
+          env:
+            # Antrea Agent needs to be aware that it is being used in EKS, as
+            # additional iptables rules may have to be installed.
+            - name: ANTREA_CLOUD_EKS
+              value: "true"
```

Comment:
Do you think it could be an enum like CLOUD_PROVIDER: EKS just in case there will be more cloud specific code?
Furthermore, could it be a config `cloudProvider` to take configuration in a consistent way? (I think the existing env variables are mainly because we have to get them from downward API) I know kube-apiserver and kubelet have this config "--cloud-provider", which might be used for same purpose.

Reply:
I'd rather keep it as an environment variable and not put it in the agent configuration. I am trying to hide this from the user. It is not really something that can be changed (although one may argue that when running on EKS the `encapMode` cannot be changed either). Basically if there was a good way to detect programmatically that the agent is running on EKS, I would be happy getting rid of this variable.

Trying to "copy" kubelet seems like a bad idea here. I feel like we are using this information for something very different, but I may be wrong. The values we use would also differ from what K8s is using, which could create confusion. They use "aws" but really this only applies to "eks".

We can move to an enum, but I am trying to avoid having to set this variable when it is not needed. I feel like if we move to an enum, then I would need to set it for GKE as well, but at the moment there is no reason to do so. I feel like if we need to do something similar for other clouds, then switching to an enum makes sense.

Reply:
OK, the explanation makes sense to me, thanks.